Component updates

its-miroma opened the pull request.

kevinthegreat1 commented:
Still breaks when I reload. I don’t know why this is happening cause I don’t know vue.

its-miroma commented:
Currently I decided to give up and wait for tomorrow morning to have a fresh mind

coz I cannot repro locally so I have to push every time and wait for it to build and it's annoying

kevinthegreat1 commented on .vitepress/theme/components/AuthorsComponent.vue:
```diff
@@ -3,98 +3,87 @@ import { useData } from "vitepress";
+      v-for="author in authors"
+      :key="author.noGitHub ? `${author.name}!` : author.name"
+      :is="author.noGitHub ? 'div' : 'a'"
+      :href="`https://github.com/${author.name}`"
+      target="_blank"
+    >
+      <img
+        :title="author.noGitHub ? options.noGitHub.replace('%s', author.name) : author.name"
+        :src="
+          author.noGitHub
+            ? '/assets/avatater.png'
+            : 'https://wsrv.nl/?af&maxage=7d&url=' +
+              encodeURIComponent(`https://github.com/${author.name}.png?size=32`)
+        "
+        :alt="author.name"
+        loading="lazy"
+        @error="onError"
+      />
+    </component>
```
From a quick look, why don't you just make it use `avatater.png` if `deletedGithub` or `noGithub`? What am I missing?

its-miroma replied:
We do not keep a list of users who have deleted their GitHub; this @error logic triggers if the image fails to load.

its-miroma committed:
rewrite update script

- ensure cwd is at project root
- don't print a message after each task is complete
- create localized versioned sidebars: `OLDVERSION-LOCALE.json`
- add warning box to all home pages (even localized)
- no longer does it have to set hardcoded latest versions
  - all the framework backend now relies on build.gradle

its-miroma committed:
rewrite transform.ts

- set static head tags in config.mts
- many head tags were broken, fixed
- dynamic head tags are now not set in `npm run dev`
- use filter in transformItems for performance

its-miroma committed:
rewrite i18n.ts

its-miroma committed:
rewrite AuthorsComponent

- fallback to avatater for deleted github accounts
- only show heading when the component is in the aside

its-miroma committed:
rewrite NotFoundComponent

- avoid using CSS classes where `<style scoped>` suffices

its-miroma committed:
rewrite versioning components

- obtain latest version from build.gradle
- use a different reminder for older versions
- use :href instead of router in VersionSwitcher
- consistently sort versions in descending order
- simplify CSS styles and logic in general

its-miroma committed:
BBB - bring banner back

the banner is now shown for non-production instances:

- locally, there's a distinction between `npm run dev` and `build`
- when coming from a Netlify Deploy Preview of a PR, shows a link to it
- detects everything based on default environment variables

its-miroma committed:
upsize logo

its-miroma committed:
rename .vitepress/config.ts

its-miroma committed:
disable versioning for PR builds

IMB11 commented on .vitepress/config.ts:
```diff
@@ -90,7 +90,7 @@ export default defineVersionedConfig(
     },
 
-    srcExclude: ["README.md"],
+    srcExclude: ["README.md", ...(typeof env === "number" ? ["versions"] : [])],
```
I think PR builds which touch versioned content or the versioning system should probably have versioning enabled no?

its-miroma replied:
I considered that: either I must call git diff with upstream (which means adding the remote in ci and fetching it), or set an environment variable depending on modified files (which I can do in GitHub Actions but idk about Netlify), or fetch the labels on the GitHub PR for if they include `new-version` (but labels are not set automatically so the first push will not include versions).
I concluded that new versions are not common enough to warrant adding all this overhead; besides with new versions, since update.ts is stable, no changes or surprises are expected and no one really reviews the old version. 
Correct me if I'm wrong, or if I've forgotten any other option.

IMB11 reviewed (dismissed):
Looks reasonable to me - lgtm if others are happy with it.

IMB11 commented:
Ah, one thing I have noticed, if we're disabling versioning for PRs we should hide the version switcher component.

its-miroma committed:
fix: remove version switching if there are none
